src/main/java/de/muenchen/isi/infrastructure/entity/Abfrage.java

```diff
@@ -48,7 +48,7 @@ public class Abfrage {
     @Column(nullable = true)
     private String bebauungsplannummer;
 
-    @Column(nullable = false)
+    @Column(nullable = false, unique = true)
```

Comment:
Die im Repository mittels findBy nach dem Namen direkt in der Datenbank gesucht wird und wegen eines fehlenden Suchindexes diese Suche sequentiell durchgeüfrht wird, sollte für das Attribute `nameAbfrage` ein Suchindex erstellt definiert werden. 

Somit wird bei vielen Entitäten die Suche erheblich beschleunigt.

```java
@Table(
        indexes = {
                @Index(name = "name_abfrage_index", columnList = "nameAbfrage")
        }
)

```

Reply:
Da es sich um ein Embeddable handelt muss der Index evtl in der Enittätsklasse definiert werden.

Reply:
Der Name der Spalte im Attribut "columnList" muss so wie der tatsächeliche Spaltenname benannt sein. Dies kannst du am besten beim Hochfahren des Backends im Profil lokal prüfen.